src/LLVM/Pretty.hs

```diff
@@ -479,17 +479,61 @@ instance PP Terminator where
 
 instance PP Instruction where
   pp = \case
-    Add {..}    -> "add"  <+> ppTyped operand0 `cma` pp operand1 <+> ppInstrMeta metadata
-    Sub {..}    -> "sub"  <+> ppTyped operand0 `cma` pp operand1 <+> ppInstrMeta metadata
-    Mul {..}    -> "mul"  <+> ppTyped operand0 `cma` pp operand1 <+> ppInstrMeta metadata
-    Shl {..}    -> "shl"  <+> ppTyped operand0 `cma` pp operand1 <+> ppInstrMeta metadata
-    AShr {..}   -> "ashr" <+> ppTyped operand0 `cma` pp operand1 <+> ppInstrMeta metadata
-    LShr {..}   -> "lshr" <+> ppTyped operand0 `cma` pp operand1 <+> ppInstrMeta metadata
+    Add {..}    -> "add"
```

Comment:
There is quite a bit of duplication going on here. What do you think about factoring this out into two helper functions, one for binary operators that have `nuw` and `nsw` flags and one operators that have an `exact` flag?

Reply:
sounds good!

Reply:
I was thinking of a helper that captures the whole instruction and takes as arguments the instruction name, the two operands, the nuw and nsw flags and the metadata. What do you think about that?

Reply:
oh, something like
`intBinOpWithWraparoundFlags :: Doc -> Bool -> Bool -> Operand -> Operand -> InstructionMetadata -> Doc`
and
`intBinOpWithExact :: Doc -> Bool -> Operand -> Operand -> Doc`?

Reply:
Exactly!

Reply:
after this most recent change, and if the build and tests pass, then I think it'll be good to go!